File: scenarios/storage-java-blob-event-notifier/golden/src/main/java/com/example/EventPublisher.java
package com.example;

import com.azure.core.exception.HttpResponseException;
import com.azure.core.util.BinaryData;
import com.azure.messaging.eventgrid.EventGridEvent;
import com.azure.messaging.eventgrid.EventGridPublisherClient;

import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;

public final class EventPublisher {
    private static final Logger LOGGER = Logger.getLogger(EventPublisher.class.getName());
    private final EventGridPublisherClient<EventGridEvent> client;

    public EventPublisher(EventGridPublisherClient<EventGridEvent> client) {
        this.client = client;
    }

    public void publish(String subject, List<DownstreamNotification> notifications) {
        List<EventGridEvent> events = notifications.stream()
                .map(notification -> new EventGridEvent(
                        subject,
                        "Contoso.Documents.Processed",
                        BinaryData.fromObject(notification),
                        "1.0"))
                .toList();
        try {
            client.sendEvents(events);
        } catch (HttpResponseException exception) {
            LOGGER.log(Level.SEVERE, "Event Grid publishing failed", exception);
            throw exception;
        }
    }
}

File: scenarios/storage-python-blob-event-notifier/golden/event_publisher.py
from __future__ import annotations

import logging
from collections.abc import Iterable
from typing import Any

from azure.core.exceptions import AzureError
from azure.eventgrid import EventGridEvent, EventGridPublisherClient
from azure.eventgrid.aio import EventGridPublisherClient as AsyncEventGridPublisherClient

logger = logging.getLogger(__name__)


def create_document_processed_event(
    data: dict[str, Any],
    *,
    subject: str = "/documents/invoices/processed",
) -> EventGridEvent:
    return EventGridEvent(
        subject=subject,
        event_type="Contoso.Documents.Processed",
        data=data,
        data_version="1.0",
    )


def publish_custom_events(
    topic_endpoint: str,
    credential: object,
    events: Iterable[EventGridEvent],
) -> None:
    client = EventGridPublisherClient(topic_endpoint, credential)
    try:
        client.send(list(events))
    except AzureError as error:
        logger.error("Custom Event Grid publishing failed: %s", error)
    finally:
        client.close()


async def publish_custom_events_async(
    topic_endpoint: str,
    credential: object,
    events: Iterable[EventGridEvent],
) -> None:
    client = AsyncEventGridPublisherClient(topic_endpoint, credential)
    try:
        await client.send(list(events))
    except AzureError as error:
        logger.error("Async custom Event Grid publishing failed: %s", error)
    finally:
        await client.close()
